### PR TITLE
Indexed lenses

### PR DESCRIPTION
**Pull request by Roxxik**
Adding Indexed Lenses 

Not sure about the name IndexedShop and 

#49

### CODE REVIEW

**Comment by paf31** [comment]
Thanks!

File: src/Data/Lens/Types.purs [hunk 1]
```diff
@@ -41,6 +43,13 @@ type Lens' s a = Lens s s a a
 type ALens s t a b = Optic (Shop a b) s t a b
 type ALens' s a = ALens s s a a
 
+-- | An indexed lens.
+type IndexedLens i s t a b = forall p. Strong p => IndexedOptic p i s t a b
+type IndexedLens' i s a = IndexedLens i s s a a
+
+type AnIndexedLens i s t a b = IndexedOptic (IndexedShop i a b) i s t a b
```
Comment (paf31) [inline src/Data/Lens/Types.purs:50]: Why is this not just `IndexedOptic (Shop a b)`?

Comment (Roxxik) [reply src/Data/Lens/Types.purs:50]: I tried it but just could not find any type safe way to get it compiling and so i tried my own shop with everything that might be needed and it worked out...

my theoretical knowledge in this regard isn't that deep...

Comment (Roxxik) [reply src/Data/Lens/Types.purs:50]: i'm not sure, i might got confused by the parameters of `Shop` being `a b s t`

Comment (Roxxik) [reply src/Data/Lens/Types.purs:50]: actually IndexedShop i a b == Shop (Tuple i a) b so IndexedShop might be removed

**Pull request merged**